### ISSUE TITLE
Add support for multinominal alignment

### ISSUE THREAD

**Issue by benbel** [liam2/liam2#268]
Currently, the `align` function and its derivatives only support alignement for binary variables (as far as I know).

It would be useful to be able to handle multinomial alignment, ie. alignement when there are more than 2 alternatives (say, for a work status variable that can take values among work, unemployment and inactivity).

There are fast and easy to implement algorithms for this such as [Bi-Proportional Scaling](https://ideas.repec.org/a/ijm/journl/v9y2016i3p89-102.html). There are [Python libraries](https://github.com/Dirguis/ipfn) that implement it.

**Comment by gdementen** [comment]
I have already implemented bi-proportional scaling several times in other projects. What is most blocking for this issue is not that algorithm but rather to integrate the functionality nicely with the rest of the framework. It should not be too hard, but it takes time and I have already my plate full. 

Release 0.13 should be based on larray where [ipfp](https://en.wikipedia.org/wiki/Iterative_proportional_fitting) (another name for bi-proportional fitting) is implemented. This will bring us one step closer to multinomial alignment. 

If you want to help to have this a bit faster (don't hold your breath), it would help if you could provide a nice simple self-contained example (along the lines of https://github.com/liam2/liam2/blob/master/liam2/tests/functional/one2many.yml)

**Comment by gdementen** [comment]
yes and if you update the documentation that would be even better :)

**Comment by benbel** [comment]
See [here](https://github.com/liam2/liam2/pull/272). Is that what you had in mind?

**Comment by gdementen** [comment]
Yes! This is a nice head start. Thanks! This is not the API I had in mind though. I wanted to have the multinomial_alignment function return the outcome directly like choice does. To credit you properly when I merge this PR (it will take at least several months before I get to it), I would like to know your name. I guess your name is your gmail address but I'd like to be sure. Also do you agree to have this contribution (and any future contribution) released under the project license (GPLv3)?

**Comment by gdementen** [comment]
Yes! This is a nice head start. Thanks! This is not the API I had in mind though. I wanted to have the multinomial_alignment function return the outcomes directly like align and choice do (*). To credit you properly when I merge this PR (it will take at least several months before I get to it), I would like to know your name (I guess your name is your gmail address but I'd like to be sure). Also do you agree to have this contribution (and any future contribution) released under the project license (GPLv3)?

(*) I also wonder if it might not be better to add that capability to the align() function.

**Comment by benbel** [comment]
I'll rewrite the test file then
